src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -66,166 +58,181 @@ public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand,
-        return String.valueOf(maxEU);
+        return String.valueOf(maxValue);
     }
 
     private void setMinValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            this.minEU = Math.min(maxEU - 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.minEU = Math.max(maxEU - 1, 0);
-        }
+        long parsedValue = GTUtility.tryParseLong(val, usePercent ? DEFAULT_MIN_PERCENT : DEFAULT_MIN_EU);
+
+        this.minValue = Math.min(this.maxValue - 1, Math.max(0, parsedValue));
     }
 
     private void setMaxValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            maxEU = Math.max(minEU + 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.maxEU = Math.max(minEU + 1, 2048);
-        }
+        long parsedValue = GTUtility.tryParseLong(val, usePercent ? DEFAULT_MAX_PERCENT : DEFAULT_MAX_EU);
+        long maxUpperLimit = usePercent ? 100 : Long.MAX_VALUE;
+
+        this.maxValue = Math.max(this.minValue + 1, Math.min(parsedValue, maxUpperLimit));
     }
 
     private boolean isInverted(){
-        return this.inverted;
+        return this.isInverted;
     }
 
     private void setInverted(boolean b){
-        this.inverted = b;
+        this.isInverted = b;
     }
 
-    @Override
-    public boolean canConnectRedstone() {
-        return true;
+    private boolean isUsePercent(){
+        return this.usePercent;
+    }
+
+    private void setUsePercent(boolean b){
+        this.usePercent =  b;
+
+        if (this.usePercent){ // using percent
+            this.minValue = DEFAULT_MIN_PERCENT;
+            this.maxValue = DEFAULT_MAX_PERCENT;
+        } else { // using discrete EU
+            this.minValue = DEFAULT_MIN_EU;
+            this.maxValue = DEFAULT_MAX_EU;
+        }
+
+        // update widgets
+        updateSyncedWidgets();
+    }
+
+    private void updateSyncedWidgets() {
+        for (Widget widget : widgetsToUpdate.widgets) {
+            ((TextFieldWidget2) widget).setPostFix(getPostFix());
+            ((TextFieldWidget2) widget).setMaxLength(getLength());
+        }
+    }
+
+    private String getPostFix(){
+        return usePercent ? " %" : " EU";
+    }
+
+    private int getLength() {
+        return usePercent ? 3 : 19;
     }
 
     @Nonnull
     @Override
     public NBTTagCompound writeToNBT(@Nonnull NBTTagCompound tagCompound) {
         super.writeToNBT(tagCompound);
-        tagCompound.setLong("maxEU", this.maxEU);
-        tagCompound.setLong("minEU", this.minEU);
+        tagCompound.setLong("maxEU", this.maxValue);
+        tagCompound.setLong("minEU", this.minValue);
         tagCompound.setInteger("outputAmount", this.outputAmount);
-        tagCompound.setBoolean("inverted", this.inverted);
-        tagCompound.setBoolean("isEnabled", this.isEnabled);
+        tagCompound.setBoolean("isInverted", this.isInverted);
```

Comment:
You write this value to `"isInverted"` but only read it using `"inverted"`. To most easily maintain backwards compat, writing and reading this value using `"inverted"` would be best. Currently this value will never be read properly for newly placed covers.

Reply:
resolved with https://github.com/GregTechCEu/GregTech/pull/1511/commits/a7bdada44f0ce242747ffa18d491df348f3774f3